internal/pointer/pointer.go

```diff
@@ -27,3 +27,23 @@ func GetOrDefault[T any](ptr *T, defaultValue T) T {
 func MakePtr[T any](value T) *T {
 	return &value
 }
+
+// MakePtrOrNilOnEmpty returns a pointer to a string when value is not empty,
+// Atlas versioned API interprets a pointer to an empty string as set.
+//
+// This creates problems in many fields, for example:
+// The API complaint if the db user password pointer goes to an empty value and
+// another auth such as X509Type is set. A nil must be sent instead to avoid issues
+//
+// https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Database-Users/operation/createDatabaseUser
+// X509type is NOT set as required in the API docs, the API complaints when it
+// is pointing to an empty value
+//
+// Generally, empty value, or more precisely, pointer to empty value is not the
+// same as unset for the API, so return nil instead of pointer to empty string
+func MakePtrOrNilOnEmpty(value string) *string {
```

Comment:
I am leaving the discussion about the usefulness of this discussion to you @josvazg and @helderjs but generally I'd prefer a generic function here:
```
func MakePtrOrNil[T comparable](value T) *T {
	if value == *new(T) {
		return nil
	}
	return &value
}
```

This works with any comparable type, see https://go.dev/play/p/72xuwRshnrw.

Reply:
Nice! Will use that instead